MonkeyWrench.Web.UI/ViewLane.aspx.cs

```diff
@@ -29,11 +30,58 @@
 public partial class ViewLane : System.Web.UI.Page
 {
 	GetViewLaneDataResponse response;
+	static Dictionary<string, string> protectedBranches = new Dictionary<string, string>();
 
 	private new Master Master
 	{
 		get { return base.Master as Master; }
 	}
+	string ParseRepo(string repo)
+	{
+		var regex  = new Regex(@"github\.com[:/](.*)$");
+		var result = regex.Match(repo).Groups[1].Value;
+		return result.Replace(".git", "");
+	}
+
+	string ParseBranch(string branch)
+	{
+		return branch.Split('/').Last();
+	}
+
+	bool IsBranchProtected(DBLane lane)
+	{
+		var repo = lane.repository;
+		var branch = lane.max_revision;
+
+		if (!repo.Contains("github")) return false;
+
+		var key = repo + ":" + branch;
+		var token = Session["github_token"];
+
+		var url = "https://api.github.com/repos/" + repo + "/branches/" + branch + "/protection";

```

Comment:
Won't this expand to something like this?

    https://api.github.com/repos/git@github.com:xamarin/xamarin-macios/branches/xcode8/protection